astropy/coordinates/representation.py

```diff
@@ -784,10 +1067,26 @@ def transform(self, matrix):
             # remaining dimensions.
             newxyz = np.einsum('...ij,j...->i...', matrix, oldxyz.value)
 
+        # Handle differentials attached to this representation
+        diffs = None
+        if self.differentials:
+            # We might be being too generous here - we could enforce that this
```

Comment:
Now that for this case we *know* these are `CartesianDifferentials`, we can simplify this.